ipfs-5-to-6/migration/config_conv.go

```diff
@@ -0,0 +1,154 @@
+package mg5
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+)
+
+func convertFile(orig string, new string, convFunc func(ds map[string]interface{}) error) error {
+	in, err := os.Open(orig)
+	if err != nil {
+		return err
+	}
+	out, err := os.Create(new)
+	if err != nil {
+		return err
+	}
+	return convert(in, out, convFunc)
+}
+
+func convert(in io.Reader, out io.Writer, convFunc func(ds map[string]interface{}) error) error {
```

Comment:
I would separate the code that does json marshaling from the code thats actually doing the conversion. Maybe have the convert method take in a map[string]interface, and return one as well, then the function that calls it handles all the json marshaling fun